### PR TITLE
Use Slack's library for calling HTTP endpoints

### PR DESCRIPTION
**Pull request by Half-Shot**
Sorry about the size of this one, but it was sort of unavoidable.

This PR changes the bridge significantly so that we now rely on `@slack/web-api` rather than doing HTTP calls ourselves, which reduces the amount of maintenance needed somewhat. API responses now also are typed and based off the slack documentation.

This PR can be treated separate from #164, but should be merged in beforehand ideally.

image

- pictured: reviewer wondering why these PRs are so massive

### CODE REVIEW

**Comment by Half-Shot** [comment]
(oh it broke, one second)

File: src/BaseSlackHandler.ts [hunk 1]
```diff
@@ -104,41 +106,47 @@ export interface ISlackFile {
...
             return;
         }
-        text = await this.replaceChannelIdsWithNames(msg, text, token);
-        return await this.replaceUserIdsWithNames(msg, text, token);
+        text = await this.replaceChannelIdsWithNames(msg, text, webClient);
+        return await this.replaceUserIdsWithNames(msg, text);
     }
 
-    public async replaceChannelIdsWithNames(message: ISlackMessage, text: string, token: string) {
+    public async replaceChannelIdsWithNames(message: ISlackMessage, text: string, slackClient: WebClient) {
```
Comment (Cadair) [inline src/BaseSlackHandler.ts]: why the mixing of `webClient` and `slackClient` shouldn't we pick one?

**Comment by anoadragon453** [comment]
image

oh god why

> +455 −1,521

File: src/BaseSlackHandler.ts (hunk 1 above)
Comment (Half-Shot) [reply src/BaseSlackHandler.ts]: oh yeah, I think I was being indecisive. probably should use slackClient

**Comment by Half-Shot** [comment]
>With all these changes the bridge is going to need significant testing before release.

This is 100% the plan.

**Pull request merged**